packages/graph/hash_graph/lib/graph/src/knowledge/entity/mod.rs

```diff
@@ -30,15 +30,27 @@ pub struct Entity {
     properties: HashMap<BaseUri, serde_json::Value>,
 }
 
+impl Entity {
+    /// Returns the properties defined in this `Entity`.
+    #[must_use]
+    pub const fn properties(&self) -> &HashMap<BaseUri, serde_json::Value> {
+        &self.properties
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::ontology::types::EntityType;
 
-    fn test_entity(json: &str) {
-        let json_value: serde_json::Value = serde_json::from_str(json).expect("invalid JSON");
+    fn test_entity(entity: &str, schema: &str) {
```

Comment:
I'm not sure how I feel about us testing the validation here..

Do we perhaps want to test this inside `entity_type` instead?

Reply:
Hmm, I think testing this here is fine as we are validating the properties. It's testing the validity of the entity, not the correctness of `EntityType::validate`

Reply:
I don't think that's quite accurate though. We are testing if the entity was deserialized correctly, we're checking if the deserialized entity validates against a schema. Those aren't equivalent, we could completely screw up our deserialization and somehow ignore all optional fields and this would still pass. That's obviously not going to happen because that's not how deserialization works but I do not think these unit tests are valuable here, and in fact I think they're somewhat misleading.

(As you say we're testing the validity of the entity, and that's nothing to do with the module, we're just testing random fake entities rather than testing the logic we've implemented directly).

Reply:
Should we then just test that the entities can be serialized and deserialized in this module, and do validations in the `entity_types` module? I think the `validate` tests would be valuable if we have test cases for the error cases as well